app/models/base.rb

```diff
@@ -27,9 +27,9 @@ def run(request_method, path, params: nil, headers: {}, body: nil)
 
 private
 
-  def post_request(path)
+  def post_request(path, attributes = {})
     post_options = {
-      body: { data: { attributes: {}, type: self.class.to_s.downcase } },
+      body: { data: { attributes: attributes, type: self.class.to_s.downcase } },
```

Comment:
I feel this will open the API for abuse with ad-hoc endpoints mutating data. I feel if https://github.com/DFE-Digital/teacher-training-api/pull/1483#discussion_r459972237 is addressed none of these changes are needed

Reply:
This endpoint doesn't mutate a resource - in essence it's triggering an event. I've added some further comments here with options - https://github.com/DFE-Digital/teacher-training-api/pull/1483#discussion_r460704748

Reply:
So in this particular case, yes, you're not mutating a resource but the capability to be able to do so is now possible through this mechanism

Reply:
I've added a new custom endpoint in order to address the concerns above - see https://github.com/DFE-Digital/publish-teacher-training/pull/1269/files#diff-72aea198f50b1d4f2407a8f9f4c904f6R15